### PR TITLE
Precompilation does not work for applications using relative path

### LINKED ISSUE #27: Precompilation does not work for applications using relative path
I have structured my ASP.NET Core project according to "Feature Slices" as outlined here:

[https://msdn.microsoft.com/en-us/magazine/mt763233.aspx](https://msdn.microsoft.com/en-us/magazine/mt763233.aspx)

The result is that the views are stored alongside the relevant models and controllers (and not in the "Views" folder). The _Layout file is stored at "Features/Shared/_Layout.cshtml".

When I run the site from Visual Studio or publish the site without razor precompilation enabled, the site works fine. When I enable precompilation I get an error in the console:

```
The layout view 'Shared/_Layout.cshtml' could not be located. The following locations were searched:
Shared/_Layout.cshtml
   at Microsoft.AspNetCore.Mvc.Razor.RazorView.GetLayoutPage(ViewContext context, String executingFilePath, String layoutPath)

```
I have changed the publish options to output the relevant cshtml files so they definitely are included in the publish destination even though I know this technically isn't required when using precompilation (this is also verified since the site works fine without the precompilation enabled).

Is having an alternate location for the Views supported in the precompilation scenario?

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #27

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @dougbu \ @Eilon

File: test/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.FunctionalTests/Resources/ApplicationConsumingPrecompiledViews.Manage.Home.Index.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <!DOCTYPE html>
 <html>
     <body>
-        AspNetCore.Areas_Manage_Views_Shared__Layout_cshtml, ClassLibraryWithPrecompiledViews.PrecompiledViews, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
+        AspNetCore._Areas_Manage_Views_Shared__Layout_cshtml, ClassLibraryWithPrecompiledViews.PrecompiledViews, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.FunctionalTests/Resources/ApplicationConsumingPrecompiledViews.Manage.Home.Index.txt]: Do we care that the generated view names are no longer CLSCompliant?

**Comment by pranavkm** [comment]
Looks like the tests went missing when I rebased. I'll send an update in a sec.

File: testapps/ApplicationUsingRelativePaths/Startup.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+        public void Configure(IApplicationBuilder app, ILoggerFactory loggerFactory)
+        {
+            loggerFactory.AddConsole();
+            app.UseMvc(routes =>
+            {
+                routes.MapRoute(
+                    name: "default",
+                    template: "{controller=Home}/{action=Index}/{id?}");
+            });
```
Comment (dougbu) [inline testapps/ApplicationUsingRelativePaths/Startup.cs]: Can this be reduced e.g. removing console logging and using `UseMvcWithDefaultRoute()`?

File: testapps/ApplicationUsingRelativePaths/Views/Home/About.cshtml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+﻿@{
+    Layout = "../Shared/_Layout.cshtml";
```
Comment (dougbu) [inline testapps/ApplicationUsingRelativePaths/Views/Home/About.cshtml]: Doesn't #33 mean this won't work?

File: testapps/ApplicationUsingRelativePaths/Controllers/HomeController.cs [hunk 4]
```diff
@@ -0,0 +1,19 @@
+﻿using Microsoft.AspNetCore.Mvc;
+
+namespace ApplicationUsingRelativePaths.Controllers
+{
+    public class HomeController : Controller
+    {
+        public IActionResult Index()
+        {
+            return View("/Views/Home/Index.cshtml");
```
Comment (dougbu) [inline testapps/ApplicationUsingRelativePaths/Controllers/HomeController.cs]: This is an absolute path. Suggest using a relative path here as well, not just for the layout.

**Pull request closed without merging**